site/src/docs/components/cookie-consent/api.mdx

```diff
@@ -0,0 +1,135 @@
+| `cookie[].id`                               | The ID of the consent. This ID is saved in the consent cookie. **Must be unique.** Only for custom cookies.  Cannot be set for a common cookie.                                                                                         |
+| `cookie[].name`                             | Readable cookie name. Only for custom cookies.  Cannot be set for a common cookie.                                                                                                                                                      |
+| `cookie[].hostName`                         | Hostname where it is used. Only for custom cookies.  Cannot be set for a common cookie.                                                                                                                                                |
+| `cookie[].description`                      | Cookie usage description. Only for custom cookies.  Cannot be set for a common cookie.                                                                                                                                                  |
+| `cookie[].expiration`                       | When does the cookie expire. Only for custom cookies. Cannot be set for a common cookie.                                                                                                                                                |
+[Table 4:Cookie information properties]|
+
+##### Callbacks
+The `content` object passed to the component can also include callback functions:
+
+| Property             | Usage                                                                                                 | Arguments                                                                                                                                                                                     |
+| -------------------- | ----------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `onAllConsentsGiven` | Called when the user has given all consents. Not called if consents are not asked for.                    | Object with cookie.id as the key and true/false as value `{matomo: true, marketing: false}`                                                                                                   |
+| `onConsentsParsed`   | Called when cookie consents are read from the browser. Called always even if consent has been given. | Object with cookie.id as the key and true/false as value `{matomo: true, marketing: false}` and boolean indicating are all consents handled. If true, there is no reason to ask for consent. |
```

Comment:
We could maybe add this: 

> Called when cookie consents are read from the browser, **before the cookie modal will possibly be shown**

Reply:
Yeah, good to know the order of function calls.